sdk/testing/src/main/java/com/mesosphere/sdk/testing/Expect.java

```diff
@@ -77,26 +77,35 @@ public void expect(ClusterState state, SchedulerDriver mockDriver) {
                 Collection<String> launchedTaskNames = new ArrayList<>();
                 // A single acceptOffers() call may contain multiple LAUNCH/LAUNCH_GROUP operations.
                 // We want to ensure they're all counted as a unit when tallying the pod.
+                Protos.ExecutorInfo launchedExecutor = null;
                 Collection<Protos.TaskInfo> launchedTaskInfos = new ArrayList<>();
                 for (Protos.Offer.Operation operation : operationsCaptor.getValue()) {
                     if (operation.getType().equals(Protos.Offer.Operation.Type.LAUNCH)) {
-                        // Old-style launch with custom executor
-                        launchedTaskNames.addAll(operation.getLaunch().getTaskInfosList().stream()
+                        // Old-style launch with custom executor in each TaskInfo
+                        Collection<Protos.TaskInfo> taskInfos = operation.getLaunch().getTaskInfosList();
+                        launchedExecutor = taskInfos.iterator().next().getExecutor();
+
+                        launchedTaskNames.addAll(taskInfos.stream()
                                 .map(task -> task.getName())
                                 .collect(Collectors.toList()));
-                        launchedTaskInfos.addAll(operation.getLaunch().getTaskInfosList().stream()
+                        launchedTaskInfos.addAll(taskInfos.stream()
                                 .map(task -> TaskPackingUtils.unpack(task))
                                 .collect(Collectors.toList()));
                     } else if (operation.getType().equals(Protos.Offer.Operation.Type.LAUNCH_GROUP)) {
-                        // New-style launch with default executor
-                        launchedTaskNames.addAll(operation.getLaunch().getTaskInfosList().stream()
+                        // New-style launch with default executor in parent LaunchGroup
+                        launchedExecutor = operation.getLaunchGroup().getExecutor();
+
+                        Collection<Protos.TaskInfo> taskInfos =
+                                operation.getLaunchGroup().getTaskGroup().getTasksList();
+
+                        launchedTaskNames.addAll(taskInfos.stream()
```

Comment:
Not critical, but I don't feel as if these two branches don't make the differences clear enough.

Reply:
Reworded the comments to be a bit clearer